src/water_cmd.cpp

```diff
@@ -477,8 +477,9 @@ CommandCost CmdBuildCanal(TileIndex tile, DoCommandFlag flags, uint32 p1, uint32
 
 		bool water = IsWaterTile(current_tile);
 
-		/* can't make water of water! */
-		if (water && (!IsTileOwner(current_tile, OWNER_WATER) || wc == WATER_CLASS_SEA)) continue;
+		/* Allow to make water of water, except in these specific situations.
+		 * The checks done via CMD_LANDSCAPE_CLEAR can provide better error messages if any. */
+		if (water && IsCanal(current_tile) && Company::IsValidID(_current_company) && (IsTileOwner(current_tile, _current_company) || GetTileOwner(current_tile) == OWNER_NONE)) continue;
```

Comment:
There are two things in this line that are weird to me.
- The `IsValidID` is not obvious why it is there. So either you have to put it in the comment why, but a few lines above they use a more verbose way: `_game_mode != GM_EDITOR`.
- For one check you  use `IsTileOwner`, for the next `GetTileOwner`. Why? Is there something wrong with using `IsTileOwner` twice? (hint: there isn't, as `IsTileOwner` executes `GetTileOwner`)